SignalShareExtension/ShareViewController.swift

```diff
@@ -544,7 +639,20 @@ public class ShareViewController: UINavigationController, ShareViewDelegate, SAE
     // Perhaps the AVFoundation APIs require some extra file system permssion we don't have in the
     // passed through URL.
     private func isVideoNeedingRelocation(itemProvider: NSItemProvider, itemUrl: URL) -> Bool {
-        guard MIMETypeUtil.utiType(forFileExtension: itemUrl.pathExtension) == kUTTypeMPEG4 as String else {
+        if ShareViewController.isSpecialItem(itemProvider:itemProvider) {
```

Comment:
Do these extra 12 lines add any value? It seems redundant and just muddies the water, but maybe there's an edge case I don't understand.

Reply:
I've streamlined this a bit, but yes - `MIMETypeUtil.utiType(forFileExtension:...)` isn't safe to call for URLs/paths without a file extension.